lib/fast_protocol.js

```diff
@@ -223,105 +212,58 @@ function fastMessageEncode(msg, default_crc_mode)
+ *             		buggy node-crc version. See the comments at the top
+ *                      of this module for more details.
  *     headerCrc        The CRC provided in the Fast message header.
  *     data             The string representation of the Fast message data.
  */
-function validateCrc(crcMode, headerCrc, data) {
-	var decodedCrcMode, error, v1CalculatedCrc, v2CalculatedCrc;
+function validateCrc(version, headerCrc, data) {
+	var valid = true;
+	var error;
 
-	switch (crcMode) {
-	case FAST_CHECKSUM_V1:
-		v1CalculatedCrc = mod_old_crc.crc16(data);
+	switch (version) {
+	case FP_VERSION_1:
+		var v1CalculatedCrc = mod_old_crc.crc16(data);
 		if (v1CalculatedCrc !== headerCrc) {
-			error = crcValidationError(crcMode, headerCrc,
-			    v1CalculatedCrc, undefined);
+			valid = false;
+			error = crcValidationError(headerCrc, v1CalculatedCrc);
 		}
 		break;
-	case FAST_CHECKSUM_V1_V2:
-		v1CalculatedCrc = mod_old_crc.crc16(data);
-		v2CalculatedCrc = mod_crc.crc16(data);
-
-		if (v1CalculatedCrc !== headerCrc &&
-		    v2CalculatedCrc !== headerCrc) {
-			error = crcValidationError(crcMode, headerCrc,
-			    v1CalculatedCrc, v2CalculatedCrc);
-		} else if (v2CalculatedCrc === headerCrc) {
-			decodedCrcMode = FAST_CHECKSUM_V2;
-		} else {
-			decodedCrcMode = FAST_CHECKSUM_V1;
-		}
-		break;
-	case FAST_CHECKSUM_V2:
-		v2CalculatedCrc = mod_crc.crc16(data);
+	default:
+		var v2CalculatedCrc = mod_crc.crc16(data);
 		if (v2CalculatedCrc !== headerCrc) {
-			error = crcValidationError(crcMode, headerCrc,
-			    undefined, v2CalculatedCrc);
+			valid = false;
+			error = crcValidationError(headerCrc, v2CalculatedCrc);
 		}
-		decodedCrcMode = FAST_CHECKSUM_V2;
-		break;
-	default:
-		mod_assertplus.fail('encountered invalid CRC mode ' + crcMode);
 		break;
 	}
 
-	return ({ decodedCrcMode: decodedCrcMode, error: error});
+	return ({ valid: valid, error: error});
 }
 
 /*
- * Return a Verror with details about a CRC validation failure based on the CRC
- * compatibility mode.
+ * Return a Verror with details about a CRC validation failure.
  *
- *     crcMode		CRC mode. Used to transition off buggy node-crc version.
- *             		See the comments at the top of this module for more
- *                      details.
  *     headerCrc        The CRC provided in the Fast message header.
- *     v1CalculatedCrc  The calculated CRC for the FAST_CHECKSUM_V1 CRC mode.
- *                      May be undefined if the CRC mode is set to
- *                      FAST_CHECKSUM_V2.
- *     v2CalculatedCrc  The calculated CRC for the FAST_CHECKSUM_V2 CRC mode.
- *                      May be undefined if the CRC mode is set to
- *                      FAST_CHECKSUM_V1.
+ *     calculatedCrc    The calculated CRC for the received message.
  */
-function crcValidationError(crcMode, headerCrc, v1CalculatedCrc,
-    v2CalculatedCrc) {
+function crcValidationError(headerCrc, calculatedCrc) {
```

Comment:
I've never looking into debugging when CRC errors show up in INC/SUP tickets. Would it help if the protocol version was in the error message?

Reply:
Good idea, will do.